index.js

```diff
@@ -37,7 +37,7 @@ HtmlWebpackPlugin.prototype.apply = function (compiler) {
 
   compiler.plugin('make', function (compilation, callback) {
     // Compile the template (queued)
-    compilationPromise = childCompiler.compileTemplate(self.options.template, compiler.context, self.options.filename, compilation)
+    compilationPromise = childCompiler.compileTemplate(self.options.template, compiler.context, 'template', compilation)
```

Comment:
The name was replaced here, because the child compiler would otherwise replace the `[hash]` by itself, causing an error. The actual provided output name for this function seems to not matter and be a temporary thing anyway. Please correct me if my assumption is wrong.